### PR TITLE
Update Go version to 1.20.3

### LINKED ISSUE #2054: publicly reported security vulnerabilities
Hi team,

In our vulnerability scan report there are 3 HIGH and 2 CRITICAL vulnerabilities in kube-state-metrics/kube-state-metrics:2.7.0

Component | Vulnerability | Severity | Installed Version | Latest Version | Description
| ---------- | ------------ | -------- | ----------------- | ---- | ---- |
| golang-runtime      | [CVE-2023-24538](https://nvd.nist.gov/vuln/detail/CVE-2023-24538) | CRITICAL | 1.19.4 | 1.20.3 | https://github.com/advisories/GHSA-v4m2-x4rp-hv22
|  golang-runtime  | [CVE-2023-24536](https://nvd.nist.gov/vuln/detail/CVE-2023-24536)   | HIGH | 1.19.4 | 1.20.3 | https://github.com/advisories/GHSA-9f7g-gqwh-jpf5
|  golang-runtime  | [CVE-2023-24534](https://nvd.nist.gov/vuln/detail/CVE-2023-24534)   | HIGH | 1.19.4 | 1.20.3 | https://github.com/advisories/GHSA-8v5j-pwr7-w5f8
|  golang-runtime | [CVE-2023-24537](https://nvd.nist.gov/vuln/detail/CVE-2023-24537)   | HIGH | 1.19.4 | 1.20.3  | https://github.com/advisories/GHSA-fp86-2355-v99r
|  etcd | [CVE-2021-28235](https://nvd.nist.gov/vuln/detail/CVE-2021-28235)  | CRITICAL | MISSING | v3.6.0-alpha.0  | https://github.com/advisories/GHSA-gmph-wf7j-9gcm

My suggestion: Golang version update to 1.20.3 should solve the problem.

### PR DESCRIPTION
**Pull request by RamakrishnanArun**
**What this PR does / why we need it**:
This PR upgrades the Go version to 1.20.3 which resolves a number of security vulnerabilities.

**How does this change affect the cardinality of KSM**: Does not change

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #2054

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -81,7 +81,10 @@ require (
 	gopkg.in/inf.v0 v0.9.1 // indirect
 	gopkg.in/ini.v1 v1.67.0 // indirect
 	gopkg.in/yaml.v2 v2.4.0 // indirect
+	k8s.io/autoscaler/vertical-pod-autoscaler v0.0.0-20191115143342-4cf961056038 // indirect
+	k8s.io/klog v1.0.0 // indirect
 	k8s.io/kube-openapi v0.0.0-20230308215209-15aac26d736a // indirect
+	k8s.io/kube-state-metrics v1.9.8 // indirect
```
Comment (mrueg) [inline go.mod:87]: Why are you adding v1 as a dependency here?

Comment (RamakrishnanArun) [reply go.mod:87]: @mrueg Thanks for your review! I think this got added when I did a `go mod tidy` and I didn't notice, my mistake. Let me remove this and push again.

Comment (RamakrishnanArun) [reply go.mod:87]: I just need to remove that one line right? (`k8s.io/kube-state-metrics v1.9.8 // indirect`)

Comment (RamakrishnanArun) [reply go.mod:87]: Actually, no, it got added when I ran `go get k8s.io/kube-state-metrics`.

**Comment by RamakrishnanArun** [comment]
Closing in favour of https://github.com/kubernetes/kube-state-metrics/pull/2056
I had issues with name/email on the commits and also cleaned up the changes as well.

**Pull request closed without merging**